release_bot/new_release.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+This module provides functionality for automation of releasing projects
+into various downstream services
+"""
+
+class New_Release:
+
+    def __init__(self):
+        # Init to default values.
+        self.changelog = None
+        self.author_name = None
+        self.author_email = None
+        self.pypi = None
+        self.trigger_on_issue = None
+        self.labels = None
+        self.pr_id = None
+        self.commitish = None
+        self.version = None
+
+    def update(self, release_conf):
```

Comment:
Where is this called/used? I don't see it anywhere.

Reply:
The `update` method is called in the [`load_release_conf`](https://github.com/Aniket-Pradhan/release-bot/blob/DictsToClasses/release_bot/releasebot.py#L65) method of `releasebot.py`.

`commitish` and `version` is used in `github.py` in [`make_new_release`](https://github.com/Aniket-Pradhan/release-bot/blob/DictsToClasses/release_bot/github.py#L301) method.